### PR TITLE
feat: on hover only show actions if available on notifications

### PR DESCRIPTION
**Pull request by rubiin**
Adds a revealer that only shows the actions when hovered over notifications
taken from aylurs dots

[Kooha-2024-10-30-05-52-14.webm](https://github.com/user-attachments/assets/52666137-03fa-481d-80fa-e7b589b4dfaa)

### CODE REVIEW

**Comment by rubiin** [comment]
Also noticed the revealer is not working as intented on OSD. The hover events doesnt do anythin atm

**Comment by Jas-SinghFSU** [comment]
Super neat, could we make this an option?

**Comment by rubiin** [comment]
doable I think.

**Comment by rubiin** [comment]
@Jas-SinghFSU added the option making it opt in

**Comment by rubiin** [comment]
@Jas-SinghFSU LGTM if there are no other changes

**Comment by Jas-SinghFSU** [comment]
Sorry, been out of town and sick the last couple of days lol. I'll get these reviewed, thanks for your efforts :heart: !

File: modules/notifications/index.ts [hunk 1]
```diff
@@ -50,15 +51,34 @@ export default (): Window<Box<Child, Attribute>, unknown> => {
...
                                 on_secondary_click: () => {
                                     notifs.CloseNotification(notif.id);
                                 },
+                                on_hover() {
+                                    if (actionsbox && showActions) actionsbox.reveal_child = true;
+                                },
+                                on_hover_lost() {
+                                    if (actionsbox && showActions) actionsbox.reveal_child = false;
+
+                                    notif.dismiss();
```
Comment (Jas-SinghFSU) [inline modules/notifications/index.ts:80]: Why are we dismissing a notification on hover loss here?

Comment (rubiin) [reply modules/notifications/index.ts:80]: Let me remove that. I guess thats not needed

**Comment by rubiin** [comment]
> Sorry, been out of town and sick the last couple of days lol. I'll get these reviewed, thanks for your efforts ❤️ !

No worries , take care of yourself as well. I had time to optimize some stuffs

**Comment by Hareekshith** [comment]
What browser is that?

**Comment by rubiin** [comment]
yeah its vivaldi

**Pull request merged**